### DISCUSSION TITLE
Can't delete invoices anymore

### DISCUSSION THREAD

**Discussion by VOOM108** [kimai/kimai#3231] [General]
It seems in the last few versions there has been a change about invoices, they can now only be cancelled, but not deleted anymore. This results in gaps in the invoice numbers. While this may be practical in some situations, it is a disadvantage in others. e.g. I must export a timesheet and an invoice for the same customer. Previously I deleted the timesheet after downloading and kept the invoice, resulting in the next invoice getting a subsequent invoice-number. Now the next invoice-number has a gap, while there never really was an invoice in between. This could be avoided by having the option to cancel OR delete an invoice. This would be a feature wish. 

For now I would like to know, if there is any way I can delete the cancelled invoices in order to get the right invoice numbers next time. I would even go to the database to achieve that, if that would not break anything.

**Comment by kevinpapst** [comment]
Even long term users (thanks for sticking with Kimai!) can learn something new from the documentation:
https://www.kimai.org/documentation/invoices.html#deleting-invoices

Really 😁

**Comment by VOOM108** [comment]
Thanks for the info.... I don't always expect documentations to be this precise and up to date, because usually with most apps they aren't. Next time I know to look there first, after something changes in the GUI ;)

**Reply by kevinpapst** [reply]
I try to keep the docs up to date for a selfish reason: the more documentation, the less support requests 😆 😈

**Comment by VOOM108** [comment]
I still have a small problem, probably being the result of impatiently deleting the invoice I wanted to cancel from the database directly. The invoice-number of the deleted entry is still blocked, meaning: kimai tries to export the invoice with that number, but then: 

`Änderungen konnten nicht gespeichert werden: Invoice "***-2022-004.docx" already exists`

Thus I cannot export the next invoice for that customer, because it won't go to ***-2022-005 - apart from changing the format of the ID altogether. Can/would you help (even though this was the result of my own stupidity)? Where except in kimai2_invoices would an invoice ID be blocked, when not deleted properly via the GUI?

**Reply by kevinpapst** [reply]
In the file system. Thats why it says ".docx". Delete it from var/invoices/.
Aside from that, you can still increase the counter manually by adding a `+1`, see docs for that.

**Reply by VOOM108** [reply]
Thanks, makes sense. I did not know that the files also reside on the server. I had solved it for now by recreating the entry in the DB and then deleting it from the GUI.
